src/shims/foreign_items.rs

```diff
@@ -948,12 +971,37 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         return Ok(None);
     }
 
-    fn eval_libc_i32(&mut self, name: &str) -> InterpResult<'tcx, i32> {
-        self
-            .eval_context_mut()
+    fn eval_libc(&mut self, name: &str) -> InterpResult<'tcx, Scalar<Tag>> {
+        self.eval_context_mut()
             .eval_path_scalar(&["libc", name])?
             .ok_or_else(|| err_unsup_format!("Path libc::{} cannot be resolved.", name).into())
-            .and_then(|scalar| scalar.to_i32())
+            .and_then(|scalar| scalar.not_undef())
+    }
+
+    fn eval_libc_i32(&mut self, name: &str) -> InterpResult<'tcx, i32> {
+        self.eval_libc(name).and_then(|scalar| scalar.to_i32())
+    }
+
+    fn set_last_error(&mut self, scalar: Scalar<Tag>) -> InterpResult<'tcx, ()> {
+        let this = self.eval_context_mut();
+        let tcx = &{ this.tcx.tcx };
+        let errno_ptr = this.machine.last_error.unwrap();
+        this.memory_mut().get_mut(errno_ptr.alloc_id)?.write_scalar(
```

Comment:
don't we have a `write_scalar` method on `InterpCx` that takes a `Pointer`?

Reply:
It takes a `Place`, should I change this?

Reply:
I guess not.

Reply:
This now encodes the size of errno in 3 places (allocation, get, set). That doesn't seem great.

I'd prefer if you could make errno an `Option<Place>` and then used the `InterpCx` methods here and for getting.

Reply:
OK, I'll change after the other changes are merged to avoid merge issues.